backend/app/mcp_server/tools/dingtalk_upload.py

```
@@ -0,0 +1,304 @@
+
+            return {
+                "attachment_id": context.id,
+                "filename": filename,
+                "size": file_size,
+                "mime_type": context.mime_type,
+                "truncated": truncation_info.is_truncated if truncation_info else False,
+            }
+
+        except requests.RequestException as e:
+            logger.error(f"Failed to download file from URL: {e}")
+            return {"error": f"Failed to download file: {str(e)}"}
```

Ruff (0.15.11)

[warning] Use explicit conversion flag

Replace with conversion flag

(RUF010)